spec/system/admin/products_v3/products_spec.rb

```diff
@@ -666,6 +670,115 @@
     end
   end
 
+  describe "Changing producers, category and tax category" do
+    let!(:variant_a1) {
+      product_a.variants.first.tap{ |v|
+        v.update! display_name: "Medium box", sku: "APL-01", price: 5.25, on_hand: 5,
+                  on_demand: false
+      }
+    }
+    let!(:product_a) {
+      create(:simple_product, name: "Apples", sku: "APL-00",
+                              variant_unit: "weight", variant_unit_scale: 1) # Grams
+    }
+
+    context "when they are under 11" do
+      before do
+        create_list(:supplier_enterprise, 9, users: [user])
+        create_list(:tax_category, 9)
+        create_list(:taxon, 2)
+
+        visit admin_products_url
+      end
+
+      it "should not display search input, change the producers, category and tax category" do
+        producer_to_select = random_producer(product_a)
+        category_to_select = random_category(variant_a1)
+        tax_category_to_select = random_tax_category
+
+        within row_containing_name(product_a.name) do
+          validate_tomselect_without_search!(
+            page, "Producer",
+            producer_to_select,
+            producer_search_selector
+          )
+        end
+
+        within row_containing_name(variant_a1.display_name) do
+          validate_tomselect_without_search!(
+            page, "Category",
+            category_to_select,
+            categories_search_selector
+          )
+          validate_tomselect_without_search!(
+            page, "Tax Category",
+            tax_category_to_select,
+            tax_categories_search_selector
+          )
+        end
+
+        click_button "Save changes"
+
+        expect(page).to have_content "Changes saved"
+        product_a.reload
+        variant_a1.reload
+
+        expect(product_a.supplier.name).to eq(producer_to_select)
+        expect(variant_a1.primary_taxon.name).to eq(category_to_select)
+        expect(variant_a1.tax_category.name).to eq(tax_category_to_select)
+      end
+    end
+
+    context "when they are over 11" do
+      before do
+        create_list(:supplier_enterprise, 11, users: [user])
+        create_list(:tax_category, 11)
+        create_list(:taxon, 11)
+
+        visit admin_products_url
+      end
+
+      it "should display search input, change the producer" do
+        producer_to_select = random_producer(product_a)
+        category_to_select = random_category(variant_a1)
+        tax_category_to_select = random_tax_category
+
+        within row_containing_name(product_a.name) do
+          validate_tomselect_with_search!(
+            page, "Producer",
+            producer_to_select,
+            producer_search_selector
+          )
+        end
+
+        within row_containing_name(variant_a1.display_name) do
+          sleep(0.1)
+          validate_tomselect_with_search!(
+            page, "Category",
+            category_to_select,
+            categories_search_selector
+          )
+          sleep(0.1)
```

Comment:
Finally, the specs work now!! 😓 
The more I wanted to avoid sleep, the more I was pushed to add it.
On my machine, without sleep (_maybe mine is already slow 😅_) , the specs are passing. But here, they were always failing. 
I tried many ways (_as you can see my so many forced pushes 😅_). But all was in vain.